### PR TITLE
PIECE: ConvertKit

### LINKED ISSUE #853: Convert Kit
Authentication: Api Key
Triggers
- New Subscriber
- New Form Subscriber
- Unsubscribe

https://developers.convertkit.com/#create-a-webhook

### PR DESCRIPTION
**Pull request by kanarelo**
## What does this PR do?

Fixes activepieces/under-review#367

### CODE REVIEW

**Comment by kanarelo** [comment]
@abuaboud Man, I am getting this error, I don't I have misconfigured anything, let me know what might be wrong

```
err: {
      "type": "TypeError",
      "message": "Cannot read properties of undefined (reading 'getTrigger')",
      "stack":
          TypeError: Cannot read properties of undefined (reading 'getTrigger')
              at getPieceTrigger (/workspace/dist/packages/backend/webpack:/activepieces/packages/backend/src/app/helper/trigger-utils.ts:158:32)
              at /workspace/dist/packages/backend/webpack:/activepieces/packages/backend/src/app/helper/trigger-utils.ts:83:26
              at Generator.next (<anonymous>)
              at /workspace/node_modules/tslib/tslib.js:118:75
              at new Promise (<anonymous>)
              at Object.__awaiter (/workspace/node_modules/tslib/tslib.js:114:16)
              at disablePieceTrigger (/workspace/dist/packages/backend/webpack:/activepieces/packages/backend/src/app/helper/trigger-utils.ts:81:118)
              at Object.<anonymous> (/workspace/dist/packages/backend/webpack:/activepieces/packages/backend/src/app/helper/trigger-utils.ts:73:19)
              at Generator.next (<anonymous>)
              at /workspace/node_modules/tslib/tslib.js:118:75
              at new Promise (<anonymous>)
              at Object.__awaiter (/workspace/node_modules/tslib/tslib.js:114:16)
              at Object.disable (/workspace/dist/packages/backend/main.js:4619:24)
              at /workspace/dist/packages/backend/webpack:/activepieces/packages/backend/src/app/instance/instance-side-effects.ts:57:76
              at Array.map (<anonymous>)
              at Object.<anonymous> (/workspace/dist/packages/backend/webpack:/activepieces/packages/backend/src/app/instance/instance-side-effects.ts:57:46)```

**Comment by kanarelo** [comment]
Happens when I am publishing

File: packages/backend/src/app/helper/trigger-utils.ts [hunk 1]
```diff
@@ -179,7 +179,7 @@ const enablePieceTrigger = async (params: EnableOrDisableParams): Promise<void>
 const getPieceTrigger = (trigger: PieceTrigger): Trigger => {
     const piece = getPiece(trigger.settings.pieceName)
 
-    if (piece === null) {
+    if (piece == null) {
```
Comment (kanarelo) [inline packages/backend/src/app/helper/trigger-utils.ts]: @abuaboud made this change to fix the bug I mentioned in this PR

File: packages/pieces/convertkit/src/lib/triggers/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    sampleData: {
+      rule: {
+        id: 3580202,
+        account_id: 1348380,
+        event: { name: 'form_subscribe', form_id: 5016504 }
+      }
+    },
+    props: {
+      // TODO change to form dropdown
+      form_id: Property.ShortText({
```
Comment (abuaboud) [inline packages/pieces/convertkit/src/lib/triggers/index.ts]: Let's change this to dynamic dropdown.

File: packages/pieces/convertkit/src/lib/triggers/index.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        "fields": {}
+      }
+    }
+  },
+  {
+    name: "form_subscribed",
+    displayName: "New Form Subscribed",
+    description: "Triggers when a subscription on a form",
+    event: "subscriber.form_subscribe",
+    sampleData: {
```
Comment (abuaboud) [inline packages/pieces/convertkit/src/lib/triggers/index.ts:38]: Can you please add sample data for all triggers

**Review by abuaboud — COMMENTED** [review]
Hi @kanarelo I updated the branch with main, and narrowed down the scope to the 3 important triggers.

Please take look at the comments and let me know what you think

**Comment by kanarelo** [comment]
@abuaboud @ashrafsamhouri  My convertkit expired after 14 days, I have tried setting up a new account, but the verification process seems not so simple; seems I may have to speak/chat to a human from their support. Kindly help get me an api key for testing.

**Pull request closed without merging**